plugins/convert_README_to_header.sh

```diff
@@ -27,11 +27,11 @@ fi
 plugin=`basename "$path"`;
 outfilename=`basename $outfile`;
 
-# See http://stackoverflow.com/a/16576291
-# On Mac OS's sed, \n is not recognized as a newline character, but
-# \[actual newline] works
-desc=`sed -e ':a' -e 'N' -e '$!ba' -e 's/\"/\\\"/g' -e 's/\n/\\\\n"\\
-"/g' "$path/README.md"`;
+#s/\"/\\\"/g - replace " with \" throughout
+#1!s/^/\"/g - Apart from the first line, add a \" to the start of each line
```

Comment:
Why do we need this? I thought the conversion here should look something like this:

```md
First line with "quotes"
Second line
```

```c
const char plugin_description[] = "First line with \"quotes\"\nSecond line";
```

Reply:
Readability/debugability/maintainability, for example if there are issues with the escaping within plugin_description its a lot easier to find them if it isn't one long continuous line. It also removes any potential for a line length limitation within a pre-processor/compiler that we're unaware of. It also means the file passes lint which I appreciate some people feel isn't necessarily important.

The current output, which my tweaked sed command matches, looks like this:
`const char plugin_description[] = "First line with \"quotes\"\n"
"Second line";`

Obviously the contents of the variable ends up like your example when interpreted by the compiler, and then when sent to the GUI it can be correctly rendered with the new lines to break up the text.

As far as I can see from here, everything used in my proposed sed command is part of the standard POSIX sed feature-set:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html

Reply:
Ah, I see. I was fooled by the comment in the next line, since it doesn't mention that a `"` is added after the `\n`. Other than that, I think it looks fine then.

Reply:
Oh yes, very true!

Reply:
Done, although I see there is also some inconsistency about escaping on the RHS of the substitution, I'll try and find out what the correct behaviour there should be.